micromanager_gui/main_window.py

```diff
@@ -152,19 +175,126 @@ def add_frame_explorer(self, name, array):
         except KeyError:
             self.viewer.add_image(array, name=layer_name)
 
-    # TO DO: add the file name form the save box
+
+    def _on_mda_started(self, sequence: useq.MDASequence):
+        """"create temp folder when mda starts."""
+
+        self.temp_folder = tempfile.TemporaryDirectory(None, str(sequence.uid))
```

Comment:
i don't see where this is ever cleaned up with `self.temp_folder.cleanup()`.  It's important that running an experiment has no "side effects" (leaving little things here and there).  So _on_mda_finished needs to call the `cleanup` method described in the [docstring](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory).

Reply:
I read about that but I checked and the temp folder that I create is deleted automatically.

I will add the cleanup() once the mda is finished.